src/ts/lib/util/common.ts

```diff
@@ -945,6 +951,79 @@ class UtilCommon {
 		return String(s || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 	};
 
+	private getPrismComponentDependencies (lang: string) {
+		const components = require(`prismjs/components.js`);
+		const result = [];
+		const language = components.languages[lang] || {};
+		// the type of `require`, `optional`, `alias` is one of `undefined`, `string[]` and `string`
+		const requirements = [].concat(language.require || []);
+		const optionals = [].concat(language.optional || []);
+		requirements.concat(optionals).forEach(item => {
+			result.push(...this.getPrismComponentDependencies(item));
+		});
+		result.push(lang);
+		return result;
+	};
+
+	/** returns an array which is the correct order of loading all Prism components */
+	prismComponents (): string[] {
+		const components = require(`prismjs/components.js`);
+		const codes = components.languages || {};
+		const result = [];
+		Object.keys(codes).forEach((key: string) => {
+			codes[key].title && result.push(...this.getPrismComponentDependencies(key));
+		});
+		return [...new Set(result)];
+	};
+
+	findRealPrismLanguageCode (lang: string){
+		if (!this.prismMap[lang]) {
+			return 'plain';
+		};
+
+		const alias = Object.keys(this.prismMap).filter(k => this.prismMap[k] == this.prismMap[lang]);
+		alias.sort();
+
+		return alias[0];
+	}
+
+	private prismAliasMap () {
+		const components = require(`prismjs/components.js`);
+		const codes = components.languages || {};
+		const result = {};
+
+		Object.keys(codes).forEach((key: string) => {
+			const lang = components.languages[key];
+			const alias = [].concat(lang.alias || []);
+			const titles = lang.aliasTitles || {};
+
+			if (lang.title) {
+				result[key] = lang.title;
+			};
+
+			alias.forEach((a: string) => {
+				result[a] = titles[a] || lang.title;
+			});
+		});
+
+		return result;
+	};
+
+	prismTitles () {
+		const valueToKeysMap: Map<string, string[]> = new Map();
```

Comment:
This method seems overcomplicated as well. It's possible to prepare all required mappings on single run without doing manipulations in run-time.